modules/core/src/test/java/org/apache/ignite/internal/client/thin/ConnectionLimitTest.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2023 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.client.thin;
+
+import org.apache.ignite.Ignite;
+import org.apache.ignite.client.IgniteClient;
+import org.apache.ignite.internal.processors.odbc.ClientListenerProcessor;
+import org.apache.ignite.mxbean.ClientProcessorMXBean;
+import org.apache.ignite.testframework.GridTestUtils;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Thin client connections limit tests.
+ */
+public class ConnectionLimitTest extends AbstractThinClientTest {
+    /**
+     * Default timeout value.
+     */
+    private static final int MAX_CONNECTIONS = 4;
+
+    /** */
+    @Test
+    public void testConnectionsRejectedOnLimitReached() throws Exception {
```

Comment:
What happens if we change the limit dynamically to a lower value? E.g.
* Set the limit to 5
* Connect 5 clients
* Set the limit to 2

Let's add a test for this scenario.

Reply:
Good idea.